src/backend/commands/trigger.c

```diff
@@ -6613,39 +6616,24 @@ void EndCompositeTriggers(bool error)
  * Find if the trigger is TSQL or PG type
  */
 static
-bool IsCompositeTrigger(Oid fn_oid)
+bool IsCompositeTrigger(Oid fn_oid, List *fn_name)
 {
 	bool	isComposite = false;
 	Oid		pltsql_lang_oid = InvalidOid;
 	Oid		pltsql_validator_oid = InvalidOid;
-	/*
-	 * FIXME: the following typedef (PLtsql_config) describes a
-	 * structure shared with PLtsql.  Since the type is shared,
-	 * we should find a header file that is properly shared and
-	 * move this typedef
-	 *
-	 * This typedef *must* be kept in-synch with the PLtsql_config
-	 * type declaration found in contrib/babelfishpg_tsql/src/pgtsql.h
-	 */
-	typedef struct PLtsql_config
-	{
-		char	*version;         /* Extension version info */
-		Oid		 handler_oid;     /* Oid of language handler function */
-		Oid		 validator_oid;   /* Oid of language validator function */
-	} PLtsql_config;
 
-	PLtsql_config **config = (PLtsql_config **) find_rendezvous_variable("PLtsql_config");
-
-	if (*config)
-	{
-		pltsql_lang_oid = (*config)->handler_oid;
-		pltsql_validator_oid = (*config)->validator_oid;
-	}
+	if (get_func_language_oids_hook)
+		get_func_language_oids_hook(&pltsql_lang_oid, &pltsql_validator_oid);
 
 	if (pltsql_lang_oid != InvalidOid)
 	{
 		HeapTuple	tuple;
 		Form_pg_proc procedureStruct;
+		/* If function oid is invalid, get it using function name */
+		if (!OidIsValid(fn_oid) && fn_name != NULL)
+		{
+			fn_oid = LookupFuncName(fn_name, 0, NULL, false);
```

Comment:
Just for my curiosity, in which case given fn_oid is invalid but we can find it by name?

Reply:
this is for create trigger case, the main function gets function id much later and all the TSQL checks are earlier. For DML cases function id is valid.